src/core/public/ui_settings/ui_settings_service.test.ts

```diff
@@ -0,0 +1,121 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+function mockClass<T>(
```

Comment:
@azasypkin Tried making a helper for mocking class again, and this time I really like how it turned out. Thoughts?

Reply:
Note the use of `.mockName()` and how that integrates with the snapshots of the classes: https://github.com/elastic/kibana/pull/22694/files#diff-de8d63302459e4f07e8b9f006649574fR17

Reply:
Paid a little extra attention to make it serialize in snapshots nicely too: https://github.com/elastic/kibana/pull/22694/files#diff-de8d63302459e4f07e8b9f006649574fR17

Reply:
I like it! We can probably move it to some shared test tools/utils and use it in other tests from now on. Just two things:

* I can't think of any real use case right now, but it may happen that file/module exports both class and something else that we'd like to mock and we won't be able to use `mockClass` for that or I'm wrong?

* Wondering if we can use [`jest-mock`'s](https://www.npmjs.com/package/jest-mock) `generateMetadata` and `generateFromMetadata` to mock all class methods at once and then set behavior/return values in setup function.

Reply:
Cool, I'll follow up with something like that if/when I need this util again and will investigate how we can support modules with more exports. I've tried using the autoMocks before but found them to be really mysterious and the docs didn't really help me understand what was going on... not opposed to them, but since I've struggled to figure them out til this point I worry they might make the tests harder to understand... but really just don't know.